src/gui/TotpExportSettingsDialog.cpp

```diff
@@ -0,0 +1,128 @@
+/*
+ *  Copyright (C) 2017 KeePassXC Team <team@keepassxc.org>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 or (at your option)
+ *  version 3 of the License.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "TotpExportSettingsDialog.h"
+#include "core/Config.h"
+#include "core/Entry.h"
+#include "gui/Clipboard.h"
+#include "gui/DatabaseWidget.h"
+#include "qrcode/QrCode.h"
+#include "totp/totp.h"
+#include "ui_TotpExportSettingsDialog.h"
+#include <QMessageBox>
+#include <QPushButton>
+
+constexpr int AUTOCLOSE_TIME = 45;
+constexpr bool AUTOCLOSE_DEFAULT = true; // enable by default
+
+TotpExportSettingsDialog::TotpExportSettingsDialog(DatabaseWidget* parent, Entry* entry)
+    : QDialog(parent)
+    , m_ui(new Ui::TotpExportSettingsDialog())
+    , m_secTillClose(AUTOCLOSE_TIME)
+    , m_entry(entry)
+    , m_timer(new QTimer(this))
+{
+    m_ui->setupUi(this);
+
+    setAttribute(Qt::WA_DeleteOnClose);
+
+    loadSettings();
+
+    connect(m_ui->buttonBox, SIGNAL(rejected()), SLOT(close()));
+    connect(m_ui->buttonBox, SIGNAL(accepted()), SLOT(copyToClipboard()));
+    connect(m_timer, SIGNAL(timeout()), this, SLOT(autoClose()));
+    connect(m_ui->enableAutoCloseCheckBox, SIGNAL(toggled(bool)), SLOT(handleAutoCloseCheckBox()));
+
+    m_ui->buttonBox->button(QDialogButtonBox::Ok)->setText(tr("Copy"));
+
+    if (m_ui->enableAutoCloseCheckBox->isChecked()) {
+        m_ui->countDown->setText(tr("Closing in %n seconds.", "", AUTOCLOSE_TIME));
+        m_timer->start(1000);
+    } else {
+        m_ui->countDown->setText(tr("Auto-close after %n seconds?", "", AUTOCLOSE_TIME));
+    }
+
+    const QUrl keyUri = totpKeyUri(entry);
+    const QrCode qrc(keyUri.toEncoded());
+    const QVariant maybeIMG = qrc.toQImage(200, 4);
```

Comment:
Is `QVariant` really necessary here? Can't it just return `null`?

Reply:
We can use "nullptr," but we would also have to manage memory manually.

Reply:
QScopedPointer<...>

Reply:
I've changed it.